src/yaml/from_yaml.cr

```diff
@@ -22,39 +29,110 @@ end
 
 def Nil.new(pull : YAML::PullParser)
   location = pull.location
-  value = pull.read_scalar
-  if value.empty?
+  case value = pull.read_plain_scalar
+  when "", "~", "null", "Null", "NULL"
     nil
   else
-    raise YAML::ParseException.new("Expected nil, not #{value}", *location)
+    raise YAML::ParseException.new("Expected null, not '#{value}'", *location)
   end
 end
 
 def Bool.new(pull : YAML::PullParser)
-  pull.read_scalar == "true"
+  location = pull.location
```

Comment:
Retrieving the location should be moved into the `else` branch, it is only relevant if an exception is created. Same for other occurrences.